storage/registers.go

```diff
@@ -0,0 +1,39 @@
+package storage
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// RegisterIndex defines methods for the register index.
+type RegisterIndex interface {
+	RegisterIndexReader
+	RegisterIndexWriter
+}
+
+// RegisterIndexReader defines read-only operations on the register index.
+type RegisterIndexReader interface {
+	// Get register by the register ID at a given block height.
+	//
+	// If the register at the given height was not indexed, returns the highest
+	// height the register was indexed at.
+	// An error is returned if the register was not indexed at all or if the height is out of bounds.
+	// Expected errors:
+	// - storage.ErrHeightNotIndexed if the register was not found in the db or is out of bounds.
```

Comment:
```suggestion
	// Expected errors:
	// - storage.ErrHeightNotIndexed if the given height was not indexed yet or lower than root height.
	// - storage.ErrNotFound if the given height is indexed, but the register does not exist.
```

Reply:
let's use `lower than first indexed height.`

Root block has a specific meaning, and overloading it here will get confusing. Especially as we add pruning to keep the last N blocks of data.